modules/mediawiki/files/bin/deploy-mediawiki.py

```diff
@@ -26,154 +111,154 @@ def _get_deployed_path(repo):
+        postinstall = []
+        stage = []
+        if not args.nolog:
+            os.system(f'/usr/local/bin/logsalmsg {text}')
+        else:
+            print(text)
+        if args.world and not args.pull:
+            pull = ['world']
+        if args.pull or args.world:
+            if args.pull:
+                pull = str(args.pull).split(',')
+            elif args.world and 'world' not in pull:
+                pull.append('world')
+            for repo in pull:
+                if repo == 'world':
+                    sm = True
+                else:
+                    sm = False
+                try:
+                    stage.append(_construct_git_pull(repo, submodules=sm))
+                except KeyError:
+                    print(f'Failed to pull {repo} due to invalid name')
+
+        for cmd in stage:  # setup env, git pull etc
+            exitcodes.append(run_command(cmd))
+        non_zero_code(exitcodes)
+        for option in options:  # configure rsync & custom data for repos
+            if options[option]:
+                if option == 'world':  # install steps for w
+                    os.chdir(_get_staging_path('world'))
+                    exitcodes.append(run_command('sudo -u www-data composer install --no-dev --quiet'))
+                    rebuild.append(f'sudo -u www-data php /srv/mediawiki/w/extensions/MirahezeMagic/maintenance/rebuildVersionCache.php --save-gitinfo --wiki={envinfo["wikidbname"]}')
+                    rsyncpaths.append('/srv/mediawiki/cache/gitinfo/')
+                rsync.append(_construct_rsync_command(time=args.ignoretime, location=f'{_get_staging_path(option)}*', dest=_get_deployed_path(option)))
+        non_zero_code(exitcodes)
+        if args.files:  # specfic extra files
+            files = str(args.files).split(',')
+            for file in files:
+                rsync.append(_construct_rsync_command(time=args.ignoretime, recursive=False, location=f'/srv/mediawiki-staging/{file}', dest=f'/srv/mediawiki/{file}'))
+        if args.folders:  # specfic extra folders
+            folders = str(args.folders).split(',')
+            for folder in folders:
+                rsync.append(_construct_rsync_command(time=args.ignoretime, location=f'/srv/mediawiki-staging/{folder}/*', dest='/srv/mediawiki/{folder}/'))
+
+        if args.extensionlist:  # when adding skins/exts
+            rebuild.append(f'sudo -u www-data php /srv/mediawiki/w/extensions/CreateWiki/maintenance/rebuildExtensionListCache.php --wiki={envinfo["wikidbname"]}')
+
+        for cmd in rsync:  # move staged content to live
+            exitcodes.append(run_command(cmd))
+        non_zero_code(exitcodes)
+        # These need to be setup late because dodgy
+        if args.l10nupdate:  # used by automated maint
+            run_command(f'sudo -u www-data ionice -c idle /usr/bin/nice -n 15 /usr/bin/php /srv/mediawiki/w/extensions/LocalisationUpdate/update.php --wiki={envinfo["wikidbname"]}')  # gives garbage errors
```

Comment:
Can this use --quiet?

Reply:
This was copied from the cron. Do we want it?

Reply:
Yes, it used to use quiet before it was a part of deploy-mediawiki.